code/UI/Interactions/InteractionMenu.cs

```diff
@@ -0,0 +1,48 @@
+﻿namespace Sports.UI;
+
+public class InteractionMenu : Panel
+{
+	private Entity Entity { get; set; }
+	private List<Interaction> InteractionList { get; set; } = new();
+
+	public Label NameLabel { get; set; }
+
+	public InteractionMenu()
+	{
+		StyleSheet.Load( "/UI/Interactions/InteractionMenu.scss" );
+
+		NameLabel = Add.Label( "placeholder", "name" );
+	}
+
+	public void SetEntity( Entity entity )
+	{
+		Entity = entity;
+		NameLabel.Text = entity.Client.Name;
+
+		foreach ( Interaction interaction in (entity as BasePlayer).GetInteractions() )
```

Comment:
For cleanliness' sake, maybe worth separating (entity as BasePlayer) into a variable above the loop.